mixins/kubernetes/rules/recording_and_alerting_rules/templates/ci_recommended_alerts.json

```diff
@@ -67,7 +67,7 @@
                     },
                     {
                         "alert": "Average Memory usage per container is greater than 95%.",
-                        "expression": "(container_memory_working_set_bytes{container!=\"\", image!=\"\", container!=\"POD\"} / on(namespace,cluster,pod,container) group_left kube_pod_container_resource_limits{resource=\"memory\", node!=\"\"}) > .95 ",
+                        "expression": "sum by (namespace, controller, container, cluster)(((container_memory_working_set_bytes{container!=\"\", image!=\"\", container!=\"POD\"} / on(namespace,cluster,pod,container) group_left kube_pod_container_resource_limits{resource=\"memory\", node!=\"\"})) * on(namespace, pod,  cluster) group_left(controller) label_replace(kube_pod_owner{owner_name=~\"(.*)\"}, \"controller\", \"$1\", \"owner_name\", \"(.*)\")) > .95",
```

Comment:
this gave false aelrts.
it's a problem to put the 'sum by' where you put it, becuase you are summing the ratio of all containers (weather there memory is over 95% or not) , and only then comparing to > 0.95, so you false results.
It can be something like that, where the aggregation function is done after the 95%. also it doesnt make sense to sum the ratio of the containers. maybe avg is better:

avg by (namespace, controller, container, cluster)(
(((container_memory_working_set_bytes{container!="", image!="", container!="POD", cluster= "azmon-prod-1p-weu-0", namespace="correlationplat"} / on(namespace,cluster,pod,container) group_left kube_pod_container_resource_limits{resource="memory", node!="",cluster= "azmon-prod-1p-weu-0", namespace="correlationplat"}))


 * on(namespace, pod,  cluster) group_left(controller) label_replace(kube_pod_owner{owner_name=~"(.*)", cluster= "azmon-prod-1p-weu-0", namespace="correlationplat"}, "controller", "$1", "owner_name", "(.*)")) > 0.95)

this comment is also true to some other places where you added sum

Reply:
Thanks for the inputs. I updated all the queries which I had put "sum by" in, according to the above. I verified the updated queries in grafana. Please let me know if they still give false alerts.